e2e/commands_test.go

```diff
@@ -31,46 +31,52 @@ services:
 # This section contains the default values for your application settings.`
 )
 
-func TestRenderBinary(t *testing.T) {
-	getDockerAppBinary(t)
-	apps, err := ioutil.ReadDir("render")
+func TestRenderTemplates(t *testing.T) {
+	skip.If(t, !hasExperimental, "experimental mode needed for this test")
+	appsPath := filepath.Join("testdata", "templates")
+	apps, err := ioutil.ReadDir(appsPath)
 	assert.NilError(t, err, "unable to get apps")
 	for _, app := range apps {
-		if app.Name() == "testdata" {
-			continue
-		}
-		t.Log("testing", app.Name())
-		envs := []string{}
+		appPath := filepath.Join(appsPath, app.Name())
 		if !checkRenderers(app.Name(), renderers) {
-			t.Log("Required renderer not enabled.")
+			t.Log("Required renderer not enabled")
 			continue
-		} else if strings.HasPrefix(app.Name(), "template-") {
-			envs = append(envs, "DOCKERAPP_RENDERERS="+strings.TrimPrefix(strings.TrimSuffix(app.Name(), ".dockerapp"), "template-"))
-		}
-		settings, overrides, env := gather(t, filepath.Join("render", app.Name()))
-		args := []string{
-			"render",
-			filepath.Join("render", app.Name()),
 		}
-		for _, s := range settings {
-			args = append(args, "-f", s)
-		}
-		for _, c := range overrides {
-			args = append(args, "-c", c)
+		t.Run(app.Name(), testRenderApp(appPath, "DOCKERAPP_RENDERERS="+app.Name()))
+	}
+}
+
+func TestRender(t *testing.T) {
+	appsPath := filepath.Join("testdata", "render")
+	apps, err := ioutil.ReadDir(appsPath)
+	assert.NilError(t, err, "unable to get apps")
+	for _, app := range apps {
+		appPath := filepath.Join(appsPath, app.Name())
+		t.Run(app.Name(), testRenderApp(appPath))
+	}
+}
+
+func testRenderApp(appPath string, env ...string) func(*testing.T) {
+	return func(t *testing.T) {
+		envs := map[string]string{}
+		data, err := ioutil.ReadFile(filepath.Join(appPath, "env.yml"))
+		assert.NilError(t, err)
+		assert.NilError(t, yaml.Unmarshal(data, &envs))
+		args := []string{"render", filepath.Join(appPath, "my.dockerapp"),
+			"-f", filepath.Join(appPath, "settings-0.yml"),
 		}
-		for k, v := range env {
+		for k, v := range envs {
 			args = append(args, "-s", fmt.Sprintf("%s=%s", k, v))
 		}
-		t.Logf("executing with %v (envs %v)", args, envs)
-		cmd := exec.Command(dockerApp, args...)
-		cmd.Env = envs
-		output, err := cmd.CombinedOutput()
-		checkResult(t, string(output), err, filepath.Join("render", app.Name()))
+		result := icmd.RunCmd(icmd.Cmd{
+			Command: append([]string{dockerApp}, args...),
```

Comment:
Why don't you put `dockerApp` directly in the args slice initialization? You will then have
```go
icmd.Cmd{
    Command: args,
...
}
```